src/python/py/models/builder.py

```diff
@@ -291,6 +291,9 @@ def make_genai_config(self, model_name_or_path, extra_kwargs, out_dir):
                 "pad_token_id": config.pad_token_id if hasattr(config, "pad_token_id") and config.pad_token_id is not None else config.eos_token_id[0] if isinstance(config.eos_token_id, list) else config.eos_token_id,
                 "type": self.model_type[ : self.model_type.find("For")].lower(),
                 "vocab_size": self.vocab_size,
+                "model_name": model_name_or_path,
```

Comment:
The `model_name` entry will contain a local path for users who build their models from models that are already on disk. Upon further examination, I think your original approach to add `model_name` to `benchmark_e2e.py` may be better since that would allow users to specify the correct model name rather than having a local path for the name.